src/python/grpcio/grpc/_channel.py

```diff
@@ -374,13 +376,20 @@ def details(self):
                 self._state.condition.wait()
             return _common.decode(self._state.details)
 
+    def debug_error_string(self):
+        with self._state.condition:
+            while self._state.debug_error_string is None:
+                self._state.condition.wait()
+            return _common.decode(self._state.debug_error_string)
+
     def _repr(self):
         with self._state.condition:
             if self._state.code is None:
                 return '<_Rendezvous object of in-flight RPC>'
             else:
-                return '<_Rendezvous of RPC that terminated with ({}, {})>'.format(
-                    self._state.code, _common.decode(self._state.details))
+                return '<_Rendezvous of RPC that terminated with ({}, {}, {})>'.format(
```

Comment:
Three is one value too many to just toss into an unlabeled tuple - reform this so that the code is labeled as the code, the details are labeled as the details, and the debug error string is labeled as the debug error string?

Consider doing this in an extracted helper function?

How long are debug error strings expected to be? Should this representation contain newlines?

Reply:
Good call, repr of rendezvous is getting ugly. I will take a shot at a formatted version.

Right now it looks like:

```
<_Rendezvous of RPC that terminated with (StatusCode.UNKNOWN, Exception calling application: , {"created":"@1526423277.168800568","description":"Error received from peer","file":"third_party/grpc/src/core/lib/surface/call.cc","file_line":1076,"grpc_message":"Exception calling application: ","grpc_status":2})>
```

Reply:
Ok now I have it in a state where the final output looks like:

```
<_Rendezvous of RPC that terminated with:
	status = "StatusCode.UNKNOWN"
	details = "Exception calling application: "
	debug_error_string = "{"created":"@1526486784.413014692","description":"Error received from peer","file":"third_party/grpc/src/core/lib/surface/call.cc","file_line":1076,"grpc_message":"Exception calling application: ","grpc_status":2}"
>
```

Is this good? If now, what output would you like to see, and I will make the changes. Could we import json, or is that not a dependency we want? The debug_error_string will be JSON formatted